### PR TITLE
Fix gauge bar animation when requestAnimationFrame is not invoked reliably

### PR DESCRIPTION
**Pull request by isoengas**
Fixes issue #20 and #29, tested on Chrome, FF & Edge

### CODE REVIEW

**Comment by isoengas** [comment]
Any advance on this PR?

**Review by x24git — APPROVED** [review]
Just noticed some issues, see below

File: libs/ngx-gauge/src/gauge/gauge.ts [hunk 1]
```diff
@@ -256,21 +256,20 @@ export class NgxGauge implements AfterViewInit, OnChanges, OnDestroy {
...
             let previousProgress = ov ? ov * unit : 0;
             let middle = start + previousProgress + displacement * progress;
 
             self._drawShell(start, middle, tail, color);
-            if (self._animationRequestID && (runtime < duration)) {
-                self._animationRequestID = window.requestAnimationFrame((timestamp) => animate(timestamp));
-            } else {
-                window.cancelAnimationFrame(self._animationRequestID);
+            if (progress === 1) {
+                cancelAnimationFrame(requestID);
```
Comment (x24git) [inline libs/ngx-gauge/src/gauge/gauge.ts]: By calling cancelAnimationFrame, this results in a series of undefined requests in the `.clearRect` function. I tested the code again and if you keep  the original if condition
`if (self._animationRequestID && (runtime < duration)) {
self._animationRequestID = window.requestAnimationFrame((timestamp) => animate(timestamp));`

but remove the `else` statement, then it still is able to draw but does not generate a slew of errors in the console.

**Review by x24git — COMMENTED** [review]
Seems your revision is causing a series of errors ever 1ms in Chrome console
image

**Comment by isoengas** [comment]
Ok, I've added a condition to avoid the exception

**Comment by stephanebiolley** [comment]
Any news of this merge?

**Comment by Robmbackus** [comment]
Simple hack to make it work

`<ngx-gauge *ngIf="gaugeValue < 0" [type]="gaugeType"
           [value]="0"
           [append]="gaugeAppendText"
           [size]="gaugeSize">
</ngx-gauge>`

`<ngx-gauge *ngIf="gaugeValue > -1" [type]="gaugeType"
           [value]="gaugeValue"
           [append]="gaugeAppendText"
            [size]="gaugeSize">
</ngx-gauge>`

`public gaugeValue: number  = -1;`

It will only render and animate the gauge once you have a value after load.

An official fix would be much cleaner though ;)

**Comment by ashish-chopra** [comment]
Fixed in v1.0.0-beta.8

**Pull request closed without merging**